Source/WebCore/platform/network/ResourceRequestBase.cpp

```diff
@@ -224,7 +224,12 @@ void ResourceRequestBase::upgradeInsecureRequestIfNeeded(URL& url, ShouldUpgrade
 
 void ResourceRequestBase::upgradeInsecureRequestIfNeeded(ShouldUpgradeLocalhostAndIPAddress shouldUpgradeLocalhostAndIPAddress, const std::optional<uint16_t>& upgradePort)
 {
-    upgradeInsecureRequestIfNeeded(m_requestData.m_url, shouldUpgradeLocalhostAndIPAddress, upgradePort);
+    auto url = m_requestData.m_url;
+    upgradeInsecureRequestIfNeeded(url, shouldUpgradeLocalhostAndIPAddress, upgradePort);
+    if (url != m_requestData.m_url) {
+        m_requestData.m_url = url;
+        setwasSchemeOptimisticallyUpgraded(true);
+    }
```

Comment:
nit - this could just be `setWasSchemeOptimisticallyUpgraded(url != m_requestData.m_url)`.

Or you could make `upgradeInsecureRequestIfNeeded()` return whether or not the scheme was upgraded to avoid copying the URL.